.github/workflows/release.yml

```diff
@@ -1,76 +1,60 @@
-# This workflows will build, test and upload the package to npm when a release is created
+# This workflows will build, test and upload the package to npm when a release
+# is created.
+# To create a release, please follow the steps documented in the "Build &
+# Publish" section of the README.md file.
 
 name: Upload Package
 on:
   push:
-    branches: [release]
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+
 jobs:
+  test:
+    uses: ./.github/workflows/end-to-end-tests.yml
+    secrets: inherit
+
   build:
     runs-on: ubuntu-20.04
+    needs: test
     steps:
       - uses: actions/checkout@v4
-      - uses: fregante/setup-git-user@v1
       # Setup .npmrc file to publish to npm
       - uses: actions/setup-node@v4
         with:
           node-version: 20
           registry-url: "https://registry.npmjs.org"
           scope: "@readalongs"
       - run: npm install
-      - name: Cypress run for web-component
-        uses: cypress-io/github-action@v6
-        with:
-          install: false
-          start: |
-            npx nx serve web-component
-            npx nx serve-test-data web-component
-          wait-on: "http://localhost:3333/build/web-component.esm.js"
-          command: npx nx test:once web-component
-      - run: cd ./packages/web-component && npm version patch
-      - run: cd ./packages/ngx-web-component && npm version patch
-      - run: npx nx build web-component && npx nx bundle web-component && npx nx build ngx-web-component
-        # We have to do this now because nx is broken (see https://github.com/nrwl/nx/issues/19989)
-      - run: |
-          export VERSION=$(npm view node_modules/@readalongs/web-component version)
-          perl -i -lne 'if (/peerDependencies/) {print; print "\"\@readalongs/web-component\": \"'$VERSION'\","} else { print }' dist/packages/ngx-web-component/package.json
-      - run: |
+      - name: Build and bundle
+        run: |
+          npx nx build web-component
+          npx nx bundle web-component
+          npx nx build ngx-web-component
+      - name: Check that the web-component bundle was up to date
```

Comment:
We could put a check here to make sure the dependency in ngx-web-component is correct as well, since we're asking people to do it manually in the README.md..

Reply:
Added this in 2604892